src/backend/distributed/commands/collation.c

```diff
@@ -0,0 +1,601 @@
+	{
+		/* no distributed types to drop */
+		return NIL;
+	}
+
+	/*
+	 * managing collations can only be done on the coordinator if ddl propagation is on. when
+	 * it is off we will never get here. MX workers don't have a notion of distributed
+	 * collations, so we block the call.
+	 */
+	EnsureCoordinator();
+
+	/*
+	 * remove the entries for the distributed objects on dropping
+	 */
+	foreach(addressCell, distributedTypeAddresses)
+	{
+		ObjectAddress *address = (ObjectAddress *) lfirst(addressCell);
+		UnmarkObjectDistributed(address);
+	}
+
+	/*
+	 * temporary swap the lists of objects to delete with the distributed objects and
+	 * deparse to an executable sql statement for the workers
+	 */
+	stmt->objects = distributedCollations;
+	char *dropStmtSql = DeparseTreeNode((Node *) stmt);
+	stmt->objects = oldCollations;
+
+	/* to prevent recursion with mx we disable ddl propagation */
+	EnsureSequentialModeForCollationDDL();
+
+	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
+								(void *) dropStmtSql,
+								ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * PlanAlterCollationOwnerStmt is called for change of ownership of collations
+ * before the ownership is changed on the local instance.
+ *
+ * If the type for which the owner is changed is distributed we execute the change on all
+ * the workers to keep the type in sync across the cluster.
+ */
+List *
+PlanAlterCollationOwnerStmt(AlterOwnerStmt *stmt, const char *queryString)
+{
+	Assert(stmt->objectType == OBJECT_COLLATION);
+
+	ObjectAddress *collationAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!ShouldPropagateObject(collationAddress))
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	QualifyTreeNode((Node *) stmt);
+	char *sql = DeparseTreeNode((Node *) stmt);
+
+	EnsureSequentialModeForCollationDDL();
+	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
+								(void *) sql,
+								ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * PlanRenameCollationStmt is called when the user is renaming the collation. The invocation happens
+ * before the statement is applied locally.
+ *
+ * As the collation already exists we have access to the ObjectAddress for the collation, this is
+ * used to check if the collation is distributed. If the collation is distributed the rename is
+ * executed on all the workers to keep the collation in sync across the cluster.
+ */
+List *
+PlanRenameCollationStmt(RenameStmt *stmt, const char *queryString)
+{
+	ObjectAddress *collationAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!ShouldPropagateObject(collationAddress))
+	{
+		return NIL;
+	}
+
+	/* fully qualify */
+	QualifyTreeNode((Node *) stmt);
+
+	/* deparse sql*/
+	char *renameStmtSql = DeparseTreeNode((Node *) stmt);
+
+	/* to prevent recursion with mx we disable ddl propagation */
+	EnsureSequentialModeForCollationDDL();
```

Comment:
comment does not match line

Reply:
this matches the comments in `commands/type.c`

Reply:
Also updated instances of this comment in `commands/type.c`